### ISSUE TITLE
Enhancement: safe shutdown

### ISSUE THREAD

**Issue by Shnitzelil** [seleniumkit/selenograph#12]
My current architecture is
Two Selenograph services behind hardware load balacner (checking the **/ping** API status code)

When I wish to restart one of the machines I would like to stop it gently...
my suggestion is to change the return status code from the  **/ping** API to be 5xx or 4xx
Or to change the return value from **ok** to something else (like "shunting down" or whatever).
Once no new connection will arrive it will be fully shutdown.

Or any suggestion you have...

WDYT?

**Comment by vania-pooh** [comment]
Seems to be a good idea. I think current implementation is here: https://github.com/seleniumkit/gridrouter/blob/master/proxy/src/main/java/ru/qatools/gridrouter/PingServlet.java because in fact Selenograph wraps GridRouter code and adds more features.

**Comment by Shnitzelil** [comment]
@vania-pooh can the grid-router can work behind load-balancer like selenograph?

**Comment by vania-pooh** [comment]
@Shnitzelil sure. The main difference is the absence of MongoDB, extended stats and UI. The main advantage - it's stateless.

**Comment by Shnitzelil** [comment]
Should this be a different API not braking the Gridrouter API...
Or, should we push this enhancement to Gridrouter and selenograph will consume this once this change will be publish?

**Comment by vania-pooh** [comment]
Summon @smecsia.

**Comment by smecsia** [comment]
@Shnitzelil Yes, we need to change the Gridrouter code to affect Selenograph this way. Send a PR to Gridrouter repository and we'll rebuild Selenograph with the newer features.
I could also advice you to consider another solution. You can use nginx behind the selenograph nodes and configure it [like this](https://lincolnloop.com/blog/pro-tip-redirecting-custom-nginx-maintenance-page). This approach would allow you to reach the same result without Gridrouter code modification.
